scripts/release/publish.ts

```diff
@@ -124,7 +124,7 @@ const publishAllPackages = async ({
   dryRun?: boolean;
 }) => {
   console.log(`📦 Publishing all packages...`);
-  const command = `yarn workspaces foreach --parallel --no-private --verbose npm publish --tolerate-republish --tag ${tag}`;
+  const command = `yarn workspaces foreach --parallel --no-private --verbose exec "yarn pack && npm publish ./package.tgz --tolerate-republish --tag ${tag} && rm package.tgz"`;
```

Comment:
I'm pretty sure this change would mean that we need to change the `YARN_NPM_AUTH_TOKEN` env var to just `NPM_TOKEN` [here](https://github.com/storybookjs/storybook/blob/valentin/migrate-version-to-use-workspace-syntax/.github/workflows/publish.yml#L107).